samples/v2/pipeline_with_gcpc_types.py

```diff
@@ -0,0 +1,75 @@
+# Copyright 2021-2022 The Kubeflow Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# import os
+import subprocess
+from typing import NamedTuple
+
+# TODO: remove this subprocess when GCPC depends on KFP v2
+GCPC_PACKAGE_PATH = 'git+https://github.com/kubeflow/pipelines@temp-gcpc-kfpv2-compatibility-branch#subdirectory=components/google-cloud'
+PACKAGES_TO_INSTALL = [GCPC_PACKAGE_PATH]
+try:
+    print(
+        subprocess.check_output(
+            f'pip install --upgrade pip && pip install {GCPC_PACKAGE_PATH}',
+            shell=True))
+except subprocess.CalledProcessError as e:
+    print(e.stderr)
+    raise
+#######
+
+from google_cloud_pipeline_components.types import artifact_types
```

Comment:
would it be possible to use Vertex Metadata class definitions instead. We are working on a proposal to remove the artifact_types.py from GPCP and directly use the definitions from Vertex SD.  see https://github.com/googleapis/python-aiplatform/blob/main/google/cloud/aiplatform/metadata/schema/google/artifact_schema.py

Reply:
Thanks, Sina. I was wondering about this.

Can you help me understand if these artifacts conform to the KFP artifact protocol? Currently, KFP requires that artifact classes implement an `__init__` that accepts `name`, `uri`, and `metadata` arguments (like [this](https://github.com/kubeflow/pipelines/blob/01c9ed9bdb3df9ddbd8b30a54eddc88f096f9c26/sdk/python/kfp/components/types/artifact_types.py#L69)). We also currently require that all artifacts, including third-party artifacts, inherit from `kfp.dsl.Artifact`, though this is a requirement we could drop (and probably ought to so that third-party libraries don't need to explicitly depend on KFP).

It's possible there is some context I'm missing here and that these Vertex artifacts do (or will soon) conform.